### ISSUE TITLE
Backup?

### ISSUE THREAD

**Issue by strarsis** [jasperf/trellis-sync#2]
This script works really nice for transferring data between WordPress installations.
Is there also an example for using this script for backing up to tar/gzip, 
ideally incrementally (important for the uploads files), with date/time stamps?
I plan to run the script in regular intervals for automated site backups 
(from production site which holds the authoritative data).

**Comment by jasperf** [comment]
Yeah that would be good. [Rsynch](https://rsync.samba.org/) could probably be used for that.  A lot on my plate though so not sure when I will have time to add the necessary commands to work on that..

**Comment by strarsis** [comment]
@jasperf: This tool uses mysql dump + rsync for dumping the mysql database contents and transferring it together with the uploaded files. If the script could be modified to just pull these files into a directory or into a tar.gzip, or even to leverage the rsync incremental backup feature.

**Comment by jasperf** [comment]
Rsync should perform incremental backups by default. If you want to automate these backups you can read about that at https://www.maketecheasier.com/make-incremental-backups-with-rsync/ . And it uses compression using the `-z` in `rsynch -az`:  `-z, --compress  compress file data during the transfer` for transfer. Perhaps we could work on a piped command to gzip on the destination machine though.

**Comment by strarsis** [comment]
@jasperf: Any news? This would be a great addition, also for automated remote backups with versioning. 🦄

**Comment by jasperf** [comment]
Been really busy working on [Stedding - Ansible playbooks for Laravel LEMP Stack](https://github.com/jasperf/stedding/). If you have a suggested implementation @strarsis please do  a PR. Thanks!

**Comment by jasperf** [comment]
You asked for three things. wp-ops covers two.

**Timestamped tar/gzip — yes.**

```bash
wp-ops files-backup    example.com production   # example_com_production_uploads_2026_08_21_14_32_07.tar.gz
wp-ops database-backup example.com production   # example_com_production_2026_08_21_14_32_07.sql.gz
```

**Incremental uploads backup — no.** `files-backup` tars the whole uploads directory each run. For a large media library on a schedule that is the wrong shape, and it is the part of your request that matters most.

**Scheduled/automated — not for backups.** wp-ops installs cron for log monitoring, not for backups.

For the two gaps, [Xilonz/trellis-backup-role](https://github.com/Xilonz/trellis-backup-role) is the better tool and has been all along — duply/duplicity, genuinely incremental, with a cron schedule and retention policy. It is a Trellis Galaxy role, so it installs via `galaxy.yml`.

Use that for scheduled incremental backups and wp-ops for on-demand snapshots and environment syncing.

One thing worth flagging since it concerns this repo directly: I am evaluating packaging the wp-ops database/uploads playbooks as a proper Trellis Galaxy role, installable from `galaxy.yml` like the role above rather than by copying scripts around. If that happens, this repo is the natural home for it — the name already describes exactly that, and un-archiving beats registering a new one. No timeline, and I would rather say that than promise a date I have missed before on this repo.

Closing here since the repo is being archived, but your request was never fully met — recording that plainly rather than marking it done.
